src/MOIWrapper.jl

```diff
@@ -0,0 +1,561 @@
+    (LQOI.Linear, LQOI.LE),
+    (LQOI.Linear, LQOI.GE),
+    (LQOI.SinVar, LQOI.EQ),
+    (LQOI.SinVar, LQOI.LE),
+    (LQOI.SinVar, LQOI.GE),
+    (LQOI.SinVar, LQOI.IV),
+    (LQOI.VecVar, MOI.Nonnegatives),
+    (LQOI.VecVar, MOI.Nonpositives),
+    (LQOI.VecVar, MOI.Zeros),
+    (LQOI.VecLin, MOI.Nonnegatives),
+    (LQOI.VecLin, MOI.Nonpositives),
+    (LQOI.VecLin, MOI.Zeros)
+]
+
+mutable struct ClpOptimizer <: LQOI.LinQuadOptimizer
+    LQOI.@LinQuadOptimizerBase
+    params::Dict{Symbol,Any}
+    ClpOptimizer(::Void) = new()
+end
+
+### Options
+
+# map option name to C function
+const optionmap = Dict(
+   :PrimalTolerance => set_primal_tolerance,
+   :DualTolerance => set_dual_tolerance,
+   :DualObjectiveLimit => set_dual_objective_limit,
+   :MaximumIterations => set_maximum_iterations,
+   :MaximumSeconds => set_maximum_seconds,
+   :LogLevel => set_log_level,
+   :Scaling => scaling,
+   :Perturbation => set_perturbation,
+   )
+# These options are set by using the ClpSolve object
+const solveoptionmap = Dict(
+   :PresolveType => set_presolve_type,
+   :SolveType => set_solve_type,
+   :InfeasibleReturn => set_infeasible_return,
+   )
+
+function setoption(m::ClpOptimizer, name::Symbol, value)
+    if haskey(optionmap, name)
+        optionmap[name](m.inner,value)
+    elseif haskey(solveoptionmap, name)
+        solveoptionmap[name](m.solveroptions,value)
+    else
+        error("Unrecognized option: $name")
+    end
+end
+
+function ClpOptimizer(;kwargs...)
+    m = ClpOptimizer(nothing)
+    m.params = Dict{String,Any}()
+    MOI.empty!(m)
+    for (name,value) in kwargs
+        m.params[Symbol(name)] = value
+    end
+    return m
+end
+
+LQOI.LinearQuadraticModel(::Type{ClpOptimizer},env) = ClpModel() 
+
+LQOI.supported_constraints(s::ClpOptimizer) = SUPPORTED_CONSTRAINTS
+LQOI.supported_objectives(s::ClpOptimizer) = SUPPORTED_OBJECTIVES
+
+"""
+    change_variable_bounds!(instance, cols::Vector{Int}, 
+        values::Vector{Float64}, senses::Vector)
+        
+Change the bounds of the variable.
+"""
+function LQOI.change_variable_bounds!(instance::ClpOptimizer, cols::Vector{Int}, 
+        values::Vector{Float64}, senses::Vector)
+    
+    upperbounds = replace_inf(get_col_upper(instance.inner))    
+    lowerbounds = replace_inf(get_col_lower(instance.inner))    
+    
+    for i in 1:length(senses)
+        if senses[i] == Cchar('U')
+            upperbounds[cols[i]] = values[i]
+        elseif senses[i] == Cchar('L')
+            lowerbounds[cols[i]] = values[i]
+        else            
+            error("sense is " * string(senses[i]) * ", but only " * Cchar('U') *  
+            " and " * Cchar('L') * " senses are supported")
+        end
+    end
+    
+    chg_column_upper(instance.inner, upperbounds)
+    chg_column_lower(instance.inner, lowerbounds)
+end
+
+"""
+    get_variable_lowerbound(instance, col::Int)::Float64
+
+Get the lower bound of the variable in 1-indexed column `col` of the model `m`.
+"""
```

Comment:
These don't need to be documented since they are just copies of the documentation in LQOI.
https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/d6957f357227e692c0e15bd6c64e9d0a37b7b1f1/src/solver_interface.jl#L121-L125